modules/base/include/modules/base/algorithm/camerautils.h

```diff
@@ -0,0 +1,190 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <modules/base/basemoduledefine.h>
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/compositeproperty.h>
+#include <inviwo/core/properties/buttonproperty.h>
+#include <inviwo/core/ports/volumeport.h>
+#include <inviwo/core/ports/meshport.h>
+#include <inviwo/core/common/inviwo.h>
+
+#include <modules/base/algorithm/mesh/axisalignedboundingbox.h>
+
+namespace inviwo {
+
+namespace camerautil {
+
+enum class Side { XNegative, XPositive, YNegative, YPositive, ZNegative, ZPositive };
+
+/**
+ * Setup the camera parameters such that the whole bounding box spaned by basisAndOffset will be
+ * inside the view frustum.
+ *
+ * @param cam       the Camera to update
+ * @param basisAndOffset the basis and offset of the bounding box that will fit inside the new view
+ * frustum
+ * @param side       this side of the bounding box will be facing the camera afterward
+ * @param fitRatio   determines the spacing between volume and the boundaries of the view frustum. A 
+ *                   fit ratio of 1 means a perfect fit, no space between frustum and volume. The aspect 
+ *                   ratio is taken into account.
+ * @param setNearFar the camera's new/far clip ranges are updated if true @see setCameraNearFar
+ * @param setLookRanges   the camera's look-to/look-from ranges are updated if true @see setCameraLookRanges
+ */
+IVW_MODULE_BASE_API void setCameraView(CameraProperty &cam, const mat4 &basisAndOffset, Side side,
```

Comment:
Suggested API changes:
- Replace side enum with a view vector in world space. The enum could encode view vectors, e.g. Side::XNegative = (-1,0,0).
- Take the bounding geometry + transformation matrix as input, then we can use it for general meshes as well

Reply:
> Replace side enum with a view vector in world space. The enum could encode view vectors, e.g. 
> Side::XNegative = (-1,0,0).

The fitting will probably not work for arbitrary vector since it works "with the side of the box face the camera". So not going to do that

> Take the bounding geometry + transformation matrix as input, then we can use it for general meshes as well 

That is encoded in the matrix

Reply:
I don’t think that it can be encoded in a matrix. The matrix assumes that the coordinates lie in texture space between [0 1] and that 0 correspond to the min point and 1 to the max from what I can understand?

Reply:
> I don’t think that it can be encoded in a matrix. The matrix assumes that the coordinates lie in texture space between [0 1] and that 0 correspond to the min point and 1 to the max from what I can understand? 

This function works for meshes aswell, see the helper class bellow.  We could consider adding an overload to this function taking a mesh/list of meshes, feel free to do that in a future PR

Reply:
I would argue against generalization and for the use of the enum with the six sides.

Reply:
I think that combining your aspect ratio code with the previous near-far computation would make it work for arbitrary geometry.

Reply:
I've added overloads for meshes now